src/peer_channels.rs

```diff
@@ -59,6 +59,9 @@ pub struct Iroh {
     /// [Gossip] needed for iroh peer channels.
     pub(crate) gossip: Gossip,
 
+    /// Sequence numbers for gossip channels.
+    pub(crate) sequence_numbers: RwLock<HashMap<TopicId, i32>>,
```

Comment:
So, `.read()` on this is called only in the tests, maybe a mutex is sufficient? Secondly, `get_and_incr()` is called right before `iroh_channels.write()` so maybe just merge this into `iroh_channels` so that it has `(ChannelState, i32)` instead of just `ChannelState`?

Reply:
Isn't it a problem if broadcasted sequence numbers are reordered for the same topic?